internal/gateway/domain_name_instance.go

```diff
@@ -352,15 +231,29 @@ func confirmDnsControllerDeployed(
 
 	// create external dns controller workload definition
 	createdWorkloadDef, err := client.CreateWorkloadDefinition(r.APIClient, r.APIServer, &externalDnsWorkloadDefinition)
-	if err != nil {
+	if err != nil && !errors.Is(err, client.ErrConflict) {
 		return fmt.Errorf("failed to create external dns controller workload definition: %w", err)
 	}
 
+	// get external dns controller workload definition id
+	var externalDnsWorkloadDefinitionId *uint
+	if !errors.Is(err, client.ErrConflict) {
+		externalDnsWorkloadDefinitionId = createdWorkloadDef.ID
+	} else {
+		existingWorkloadDef, err := client.GetWorkloadDefinitionByName(r.APIClient, r.APIServer, workloadDefName)
+		if err != nil {
+			return fmt.Errorf("failed to get existing workload definition: %w", err)
+		}
+		externalDnsWorkloadDefinitionId = existingWorkloadDef.ID
+	}
+
 	// create external dns workload instance
 	externalDnsWorkloadInstance := v0.WorkloadInstance{
-		Instance:                    v0.Instance{Name: &workloadDefName},
+		Instance: v0.Instance{
+			Name: util.StringPtr(fmt.Sprintf("%s-%s", workloadDefName, *kubernetesRuntimeInstance.Name)),
+		},
```

Comment:
Append `KubernetesRuntimeInstance` name to `external-dns` for each instance to avoid name collisions in the Threeport API.